src/main.cpp

```diff
@@ -1,9 +1,50 @@
 #include <cstdlib>
 #include <iostream>
+#include <boost/program_options.hpp>
+#include <iterator>
+
+#include "parse.h"
+#include "compiling.h"
+
+using namespace std;
+namespace po = boost::program_options;
+
+
 
 int main(int argc, char *argv[])
 {
-    std::cout << "argc == " << argc << '\n';
+    // cout << "argc == " << argc << '\n';
+
+    try {
+        // Add options and their descriptions
+        po::options_description desc("Options");
+        desc.add_options()
+            ("help,h", "List of available commands")
+            ("version", "Current version")
+            ("build", "Build and run the program");
+
+        // Get the input from the command line and store it in a variable map
+        po::variables_map vm;
+        po::store(parse_command_line(argc, argv, desc), vm);
+        po::notify(vm);
+
+        // Output based on command
+        if (vm.count("help"))
+            std::cout << desc << '\n';
+        else if (vm.count("version"))
+            // std::cout << "Current version: " << vm["age"].as<float>() << '\n';
+            std::cout << "Current version: 0.1" << '\n';
+        else if (vm.count("build"))
+            std::cout << "And this is where I would put my build functions..." << '\n';
+    }
+
+    catch(exception& e) {
+        cerr << "error: " << e.what() << "\n";
+        return 1;
+    }
+    catch(...) {
+        cerr << "Exception of unknown type!\n";
+    }
 
-    return EXIT_SUCCESS;
+    return 0;
```

Comment:
It's generally bad practice to have "magic numbers" in code. Upon first read, it's not obvious what meaning "0" has. Please use a predefined constant such as `EXIT_SUCCESS` (from `cstdlib`) or, if you need a status that's not predefined, define it as a macro at the top of the program.

Reply:
Looking at this closer, you should probably return the result of whatever option the user chooses (i.e. compiling)

Reply:
So it should just be a string describing what is happening?